api/v1/nodescan/service.go

```diff
@@ -273,17 +276,31 @@ func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeV
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Handle the new format of the request and scan node inventory additionally
+	// Scan Components containing pkgs from NodeInventory
 	if req.GetComponents() != nil {
-		layer, err := apiV1.GetVulnerabilitiesForComponents(s.db, req.GetComponents(), common.HasUncertifiedRHEL(req.GetNotes()))
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+		if resp.Features, err = s.getNodeInventoryVulns(req.GetComponents(), common.HasUncertifiedRHEL(req.GetNotes())); err != nil {
+			return nil, err
 		}
-		resp.Features = features.ConvertFeatures(layer.Features)
 	}
 	return resp, nil
 }
 
+func (s *serviceImpl) getNodeInventoryVulns(components *v1.Components, isUncertifiedRHEL bool) ([]*v1.Feature, error) {
+	log.Debugf("Scanning NodeInventory")
+	// Convert content sets to CPEs
+	cpes := s.repoToCPE.Get(components.GetRhelContentSets())
+	log.Debugf("Converted content sets '%v' to CPEs '%v'", components.GetRhelContentSets(), cpes)
+	for _, comp := range components.GetRhelComponents() {
+		comp.Cpes = append(comp.Cpes, cpes...)
```

Comment:
Maybe ensure `comp.Cpes` is empty. Or were you purposefully appending not to overwrite?

We will revisit this piece of code as part of ROX-14414. By then, we will have to decide how to support input that has CPEs set and inputs that don't. But until then, overwriting to sanitize input seems the right course of action.

Reply:
Okay, I was appending not to overwrite - having more CPEs seems not bad to me. But if we have a followup to tackle this, then I will follow your suggestion too overwrite and leave a comment about the follow-up ticket.